test/functional/test_framework/messages.py

```diff
@@ -1050,13 +1050,11 @@ def deserialize(self, f):
 
         self.nStartingHeight = struct.unpack("<i", f.read(4))[0]
 
-        if self.nVersion >= 70001:
-            # Relay field is optional for version 70001 onwards
-            try:
-                self.nRelay = struct.unpack("<b", f.read(1))[0]
-            except:
-                self.nRelay = 0
-        else:
+        # Relay field is optional for version 70001 onwards
+        # But, unconditionally check it to match behaviour in bitcoind
+        try:
```

Comment:
```suggestion
```

Looks like you still treat it as optional, no?

Reply:
On a second thought,  I don't think this generally works the way you want it to work. (See my latest comment on your other pull request: https://github.com/bitcoin/bitcoin/pull/19509#issuecomment-729462566 )

So I've unresolved this conversation again.